src/libraries/Microsoft.Extensions.Hosting/tests/UnitTests/Internal/HostTests.cs

```diff
@@ -1371,6 +1470,29 @@ public ValueTask DisposeAsync()
             }
         }
 
+        private class TestBackgroundService : BackgroundService
```

Comment:
Looking deeper, I don't think the test added is testing the same scenario I had in mind. The reason is because the base BackgroundService.StartAsync doesn't wait for the `ExecuteAsync` to complete. It returns to the Host.

Instead, I think we should switch this to not inherit from `BackgroundService`, and instead implement `IHostedService` directly and in `StartAsync` we should `await` the `emulateWorkTask`.

And then we need to switch the services around in the test, so the `AsyncThrowingService` gets started first. Then this service gets started. Meanwhile in the background, `AsyncThrowingService` throws an exception, which causes `StopApplication` to be called while this service is starting up.

Reply:
Done, pushed the updated test case.